spec.html

```diff
@@ -33815,29 +33820,32 @@ <h1>String.prototype.padStart ( _maxLength_ [ , _fillString_ ] )</h1>
         <p>When the `padStart` method is called, the following steps are taken:</p>
         <emu-alg>
           1. Let _O_ be ? RequireObjectCoercible(*this* value).
-          1. Return ? StringPad(_O_, _maxLength_, _fillString_, ~start~).
+          1. Let _S_ be ? ToString(_O_).
+          1. Let _maxLen_ be ℝ(? ToLength(_maxLength_)).
+          1. Let _fillStr_ be ? ToString(_fillString_).
+          1. If _fillStr_ is *undefined*, let _filler_ be ~empty~.
+          1. Else, let _filler_ be _fillStr_.
+          1. Return StringPad(_S_, _maxLen_, _filler_, ~end~).
         </emu-alg>
 
         <emu-clause id="sec-stringpad" type="abstract operation">
           <h1>
             StringPad (
-              _O_: an ECMAScript language value,
-              _maxLength_: an ECMAScript language value,
-              _fillString_: an ECMAScript language value,
+              _S_: a String
+              _maxLength_: a non-negative integer that is less than 2 ** 53
+              _fillString_: a String or ~empty~
               _placement_: ~start~ or ~end~,
-            ): either a normal completion containing a String or an abrupt completion
+            ): a String
           </h1>
           <dl class="header">
           </dl>
           <emu-alg>
-            1. Let _S_ be ? ToString(_O_).
-            1. Let _intMaxLength_ be ℝ(? ToLength(_maxLength_)).
             1. Let _stringLength_ be the length of _S_.
-            1. If _intMaxLength_ &le; _stringLength_, return _S_.
-            1. If _fillString_ is *undefined*, let _filler_ be the String value consisting solely of the code unit 0x0020 (SPACE).
-            1. Else, let _filler_ be ? ToString(_fillString_).
+            1. If _maxLength_ &le; _stringLength_, return _S_.
+            1. If _fillString_ is ~empty~, let _filler_ be the String value consisting solely of the code unit 0x0020 (SPACE).
```

Comment:
I'd inline this step into `padStart`/`padEnd` as well, I think.

Reply:
tbh it feelsbadman to have all these steps repeated in both padEnd and padStart. what about another non-pure AO that they can both share, that calls into the pure StringPad?

Reply:
Ehhhh I think in this case I would lean towards keeping the duplication, personally. I am a lot less inclined to factor out the "type-coerce / provide defaults for all the arguments" steps than I am most logic.